ContinueWatching/ContinueWatching.cpp

```diff
@@ -369,11 +370,46 @@ namespace WPEFramework {
 
 				result = SecKey_GetInstance(sec_proc, mSecObjectId, &sec_key);
 				if(result != SEC_RESULT_SUCCESS)
-					throw "Failure to get SecAPI key handle!";
+				{
+					result = SecKey_Generate(sec_proc, mSecObjectId, SEC_KEYTYPE_AES_128,loc);
+					if(result != SEC_RESULT_SUCCESS)
+					{
+						throw "Failure to generate new key! on Sec Instance";
+					}
+					else
+					{
+						result = SecKey_GetInstance(sec_proc, mSecObjectId, &sec_key);
+					
+						if(result != SEC_RESULT_SUCCESS)
+							throw "Failure to get SecAPI key handle!";
+					}
+				}
 
 				result = SecCipher_GetInstance(sec_proc, algorithm, mode, sec_key, iv, &sec_cipher);
 				if(result != SEC_RESULT_SUCCESS)
-					throw "Failure to get SecAPI cipher handler!";
+				{
+
+					char buf[64];
+					snprintf(buf, sizeof(buf), "%016llX.*", mSecObjectId);
+					std::string cmd = "rm /opt/drm/servicemgr/";
```

Comment:
Checking why do we need to delete this folder? Is this used only for ContinueWatching?